pkg/util/azureclient/azuresdk/azblob/blobs.go

```diff
@@ -0,0 +1,97 @@
+package azblob
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/bloberror"
+
+	"github.com/openshift/installer-aro-wrapper/pkg/util/azureclient"
+)
+
+// BlobsClient is a minimal interface for Azure BlobsClient
+type BlobsClient interface {
+	DownloadStream(ctx context.Context, containerName string, blobName string, o *azblob.DownloadStreamOptions) (azblob.DownloadStreamResponse, error)
+	UploadBuffer(ctx context.Context, containerName string, blobName string, buffer []byte, o *azblob.UploadBufferOptions) (azblob.UploadBufferResponse, error)
+	DeleteBlob(ctx context.Context, containerName string, blobName string, o *azblob.DeleteBlobOptions) (azblob.DeleteBlobResponse, error)
+	BlobExists(ctx context.Context, container string, blobPath string) (bool, error)
+	DeleteContainer(ctx context.Context, container string) error
+}
+
+type blobsClient struct {
+	*azblob.Client
+}
+
+var _ BlobsClient = &blobsClient{}
+
+// NewBlobsClientUsingSAS creates a new BlobsClient using SAS
+func NewBlobsClientUsingSAS(ctx context.Context, sasURL string, environment *azureclient.AROEnvironment) (*blobsClient, error) {
+	options := &azblob.ClientOptions{
+		ClientOptions: azcore.ClientOptions{
+			Cloud: environment.Cloud,
+		},
+	}
+	client, err := azblob.NewClientWithNoCredential(sasURL, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return &blobsClient{
+		Client: client,
+	}, nil
+}
+
+// NewBlobsClientUsingSAS creates a new BlobsClient without using SAS(Only Managed Identity Clusters)
+func NewBlobsClient(ctx context.Context, environment *azureclient.AROEnvironment, serviceURL string, credential azcore.TokenCredential) (*blobsClient, error) {
```

Comment:
comment needs to be changed.  

Also if we're not using SAS, maybe change function signature to `NewBlobsClientUsingLocalAuth` ? Or whatever it's called.

Reply:
Changed the name to `NewBlobsClientUsingEntra`